libs/@local/hashql/syntax-jexpr/src/parser/array/snapshots/hashql_syntax_jexpr__parser__array__visit__tests__unclosed_array.snap

```diff
@@ -0,0 +1,14 @@
+---
+source: libs/@local/hashql/syntax-jexpr/src/parser/array/visit.rs
+description: Unclosed array should fail
+expression: "[1, 2"
+---
+[31m[lexer::lexer::unexpected-eof] Error:[0m Lexer
+   ╭─[ <unknown>:1:6 ]
+   │
+ 1 │ [1, 2
+   │      │ 
+   │      ╰─ Unexpected end of file
+   │ 
+   │ Help: Make sure all expressions, brackets, and string literals are properly closed
+───╯
```

Comment:
I guess this is the best we can do here? (Absolutely fine with an EOF error)

Reply:
I'll defer this to another PR (as this is already a very chonky) one, but actually no! There are ways to - very easily - make the diagnostic better here.

I wanted to do this anyway, but basically the idea is that we add a new method `advance_require` or `advance_expect` which takes a `SyntaxKindSet` of allowed tokens. (the normal `advance` would basically just be `advance_require` with `SyntaxKindSet::ALL`. If that token is not present we simply error out, even if EOF. That way the error message would change from: `Unexpected end of file` to `Unexpected end of file, expected ]`. Which is a lot friendlier.